python/LoggingWrapper.cpp

```diff
@@ -49,91 +46,27 @@ namespace {
-            break;
-        case LogLevel::warn:
-            FO_LOGGER(LogLevel::warn, python)  << python_logger << boost::log::add_value("SrcFilename", filename) << boost::log::add_value("SrcLinenum", lineno) << " : " << msg;
-            break;
-        case LogLevel::error:
-            FO_LOGGER(LogLevel::error, python) << python_logger << boost::log::add_value("SrcFilename", filename) << boost::log::add_value("SrcLinenum", lineno) << " : " << msg;
-            break;
-        }
-    }
-
-    void PythonLoggerWrapper(const LogLevel log_level, const std::string& msg,
-                             const std::string& logger_name, const std::string& filename,
-                             /*const std::string& function_name,*/ const std::string& lineno)
+    template<LogLevel log_level>
+    void PythonLoggerWrapper(const std::string& msg, const std::string& filename,
+                             const int lineno)
     {
         static std::stringstream log_stream("");
-        send_to_log(log_stream, msg,
-                    std::bind(&PythonLogger, std::placeholders::_1,
-                              log_level, logger_name, filename, /*function_name,*/ lineno));
-    }
-
-
-    // debug/stdout logger
-    void PythonLoggerDebug(const std::string& msg, const std::string& logger_name,
-                           const std::string& filename, const std::string& function_name, const std::string& lineno)
-    {
-        PythonLoggerWrapper(LogLevel::debug, msg, logger_name, filename, /*function_name,*/ lineno);
-    }
-
-
-    // info logger
-    void PythonLoggerInfo(const std::string& msg, const std::string& logger_name,
-                          const std::string& filename, const std::string& function_name, const std::string& lineno)
-    {
-        PythonLoggerWrapper(LogLevel::info, msg, logger_name, filename, /*function_name,*/ lineno);
-    }
-
-    // warn logger
-    void PythonLoggerWarn(const std::string& msg, const std::string& logger_name,
-                          const std::string& filename, const std::string& function_name, const std::string& lineno)
-    {
-        PythonLoggerWrapper(LogLevel::warn, msg, logger_name, filename, /*function_name,*/ lineno);
-    }
-
-    // error logger
-    void PythonLoggerError(const std::string& msg, const std::string& logger_name,
-                           const std::string& filename, const std::string& function_name, const std::string& lineno)
-    {
-        PythonLoggerWrapper(LogLevel::error, msg, logger_name, filename, /*function_name,*/ lineno);
+        auto logger_func = [&](const std::string& arg) {
+            FO_LOGGER(log_level, python) << boost::log::add_value("SrcFilename", filename)
+                                         << boost::log::add_value("SrcLinenum", lineno)
+                                         << arg;
+        };
+        send_to_log(log_stream, msg, logger_func);
```

Comment:
When is `logger_func` called? Are you sure you want to capture `filename` by reference? Capturing everything you need explicitly might be clearer in this case given the macro use...

Reply:
`send_to_log` splits up the python logger input into lines and uses the logger func to call back onto the boost::log logger.

https://github.com/freeorion/freeorion/pull/2909/files/67c4d8f9d33d90bbd997306763d09dc9a2bb6edd#diff-518a46164dacf94517c4776f1d790ab0R30